### DISCUSSION TITLE
Octomap update using environment_monitor

### DISCUSSION THREAD

**Discussion by ahedfares** [tesseract-robotics/tesseract_ros#83] [Q&A] [answered]
Hello,
I am trying to use octomap to update surrounding environment in tesseract scene, i am using an environment_monitor node which i ran using "UPDATE_GEOMETRY" EnvironmentUpdateType but whenever i have to send the update i have to add two commands in the request (remove_link, and add_link), this seems to be very consuming as the link´s geometry is not actually being updated but it is being deleted and readded. Is there a way to update my octomap through the environment monitor without the need to delete and add the link in every received pointcloud msg?
I have been reading through the code but found no section related to UPDATE_ENVIRONMENT in the source code.
@Levi-Armstrong 

Thanking you in advance.

**Comment by Levi-Armstrong** [comment, marked as answer]
I agree, the ocomap is a special case where you would want to update the geometry of a link. This is a planned feature, but it should not take a lot to get an initial implementation. I would start with adding UpdateLinkOctomapCommand to the environment and propagate this through handling how to update the scene graph and contact managers.   May need to discuss how to handle when a link has two octomap collision geometries to identify which one gets updated, but to begin it could be setup to only handle only links which have a single octomap.
